### PR TITLE
hiop: require RAJA when cuda is enabled

### LINKED ISSUE #675: RAJA must be used when CUDA is selected.
spack formula `hiop@develop+cuda`  failed due to [this section](https://github.com/LLNL/hiop/blob/2378fde5cc371047227c396ddaaf58b6453f928c/src/LinAlg/LinAlgFactory.cpp#L357-L373) requires RAJA (but it is within the section without using RAJA).

@cameronrutherford While I will work on updating the CMake file in hiop, we also need to update the spack formula, to activate RAJA when cuda is selected.

### PR DESCRIPTION
**Pull request by cameronrutherford**
Closes https://github.com/LLNL/hiop/issues/675

cc @nychiang @cnpetra

### CODE REVIEW

**Comment by cameronrutherford** [comment]
@spackbot fix style

**Comment by cameronrutherford** [comment]
Good to merge IMO once a maintainer gets to this

**Review by alalazo — APPROVED** [review]

**Pull request merged**
